include/nuttx/nuttx.h

```diff
@@ -48,4 +48,20 @@
 #define container_of(ptr, type, member) \
   ((type *)((uintptr_t)(ptr) - offsetof(type, member)))
 
+/* Name: is_config_set
+ *
+ * Description:
+ *   Evaluate a kconfig-provided boolen type configuration setting
+ *   at compile time
+ *
+ * Arguments:
+ *   config - The configuration setting name.
+ */
+
+#define is_config_set(config) _is_config_set(config)
```

Comment:
can we simplify the check?

Reply:
Shall I simplify this macro or simplify the whole sanity check method? Could you give me some more hints pls?

Reply:
I mean the macro self, it's specially vararg in _is_config_set_evaluate.

Reply:
Let me think a little a bit about it as well

Reply:
> I mean the macro self, it's specially vararg in _is_config_set_evaluate.

Right, I agree this is a bit hacky, lemme think of some other ways.

Reply:
Maybe we can add in `drivers/syslog/syslog_channel.c` something like:
```
#ifdef CONFIG_SYSLOG_DEFAULT
#  define SYSLOG_DEFAULT_AVAILABLE 1
#else
#  define SYSLOG_DEFAULT_AVAILABLE 0
#endif

#ifdef CONFIG_RAMLOG_SYSLOG
#  define RAMLOG_SYSLOG_AVAILABLE 1
#else
#  define RAMLOG_SYSLOG_AVAILABLE 0
#endif

#ifdef CONFIG_SYSLOG_RPMSG
#  define SYSLOG_RPMSG_AVAILABLE 1
#else
#  define SYSLOG_RPMSG_AVAILABLE 0
#endif

#ifdef CONFIG_SYSLOG_RTT
#  define SYSLOG_RTT_AVAILABLE 1
#else
#  define SYSLOG_RTT_AVAILABLE 0
#endif

#define SYSLOG_NCHANNELS (SYSLOG_DEFAULT_AVAILABLE +\
                    CONFIG_RAMLOG_SYSLOG + \
                    CONFIG_SYSLOG_RPMSG + \
                    SYSLOG_RTT_AVAILABLE)
```
??
Still some more code, but is simple that should fit to solve the issue.

Reply:
> Maybe we can add in `drivers/syslog/syslog_channel.c` something like:
> 
> ```
> #ifdef CONFIG_SYSLOG_DEFAULT
> #  define SYSLOG_DEFAULT_AVAILABLE 1
> #else
> #  define SYSLOG_DEFAULT_AVAILABLE 0
> #endif
> 
> #ifdef CONFIG_RAMLOG_SYSLOG
> #  define RAMLOG_SYSLOG_AVAILABLE 1
> #else
> #  define RAMLOG_SYSLOG_AVAILABLE 0
> #endif
> 
> #ifdef CONFIG_SYSLOG_RPMSG
> #  define SYSLOG_RPMSG_AVAILABLE 1
> #else
> #  define SYSLOG_RPMSG_AVAILABLE 0
> #endif
> 
> #ifdef CONFIG_SYSLOG_RTT
> #  define SYSLOG_RTT_AVAILABLE 1
> #else
> #  define SYSLOG_RTT_AVAILABLE 0
> #endif
> 
> #define SYSLOG_NCHANNELS (SYSLOG_DEFAULT_AVAILABLE + \
>                           RAMLOG_SYSLOG_AVAILABLE + \
>                           SYSLOG_RPMSG_AVAILABLE + \
>                           SYSLOG_RTT_AVAILABLE)
> ```
> 
> ?? Still some more code, but is simple that should fit to solve the issue.
> 
> Then I think we even should not need a static assert, but can
> 
> ```
> #if SYSLOG_NCHANNELS > CONFIG_SYSLOG_MAX_CHANNELS
> #  error "Maximum channel number exceeds."
> #endif
> ```

Oh I see. That's quite straightforward, will change to that instead. Thx!